Tests/FlatMapFirstTests.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  FlatMapFirstTests.swift
+//  CombineExtTests
+//
+//  Created by Martin Troup on 22/03/2022.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+#if !os(watchOS)
+import Combine
+import CombineSchedulers
+import Foundation
+import XCTest
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+class FlatMapFirstTests: XCTestCase {
+    var cancellables: Set<AnyCancellable>!
+
+    override func setUp() {
+        super.setUp()
+
+        cancellables = []
+    }
+
+    struct TestError: Error, Equatable {}
+
+    func test_flatMapFirst_single_upstream_single_flatMap() {
+        let testScheduler = DispatchQueue.test
+
+        var innerPublisherSubscriptionCount = 0
+        var innerPublisherCompletionCount = 0
+        var isUpstreamCompleted = false
+
+        Just("").setFailureType(to: Never.self)
+            .delay(for: 1, scheduler: testScheduler)
+            .flatMapFirst { _ -> AnyPublisher<Date, Never> in
+                return Just(Date())
+                    .delay(for: 1, scheduler: testScheduler)
+                    .handleEvents(
+                        receiveSubscription: { _ in innerPublisherSubscriptionCount += 1 },
+                        receiveCompletion: { _ in innerPublisherCompletionCount += 1 }
+                    )
+                    .eraseToAnyPublisher()
+            }
+            .sink(
+                receiveCompletion: { completion in
+                    if case .finished = completion {
+                        isUpstreamCompleted = true
+                    }
+                },
+                receiveValue: { _ in }
+            )
+            .store(in: &cancellables)
+
+        testScheduler.advance(by: 2)
+
+        XCTAssertEqual(innerPublisherSubscriptionCount, 1)
+        XCTAssertEqual(innerPublisherCompletionCount, 1)
+        XCTAssertTrue(isUpstreamCompleted)
+    }
+
+    func test_flatMapFirst_error_upstream_skipping_flatMap() {
```

Comment:
Can we change the naming to not use underscores? 
I believe that's not really common practice in Swift. 

Could also be a bit denser. Since we are in flatMapFirstTests, there's no need for `test_flatMapFirst` prefix. You can just call it `testErrorSkipsFlatmap`, or however you'd like. (This applies to the other tests too)

Reply:
I got it renamed!

Reply:
seems likje a minor conflict to fix and then we can merge :) thank you.

Reply:
My bad, sorry. I resolved it.